ddtrace/context.py

```diff
@@ -123,11 +128,45 @@ def sampling_priority(self, value):
     @property
     def _traceparent(self):
         # type: () -> str
-        if self.trace_id is None or self.span_id is None:
-            return ""
+        tp = self._meta.get(_TRACEPARENT_KEY)
+        if self.span_id is None or self.trace_id is None:
+            # if we only have a traceparent then we'll forward it
+            # if we don't have a span id or trace id value we can't build a valid traceparent
+            return tp or ""
+
+        # determine the trace_id value
+        if tp:
+            # grab the original traceparent trace id, not the converted value
+            trace_id = tp.split("-")[1]
+        else:
+            trace_id = "{:032x}".format(self.trace_id)
 
         sampled = 1 if self.sampling_priority and self.sampling_priority > 0 else 0
-        return "00-{:032x}-{:016x}-{:02x}".format(self.trace_id, self.span_id, sampled)
+        return "00-{}-{:016x}-{:02x}".format(trace_id, self.span_id, sampled)
+
+    @property
+    def _tracestate(self):
+        # type: () -> str
+        dd = _w3c_format_known_propagated_tags(self)
+
+        # get and format all other _dd.p values out of meta since we need to propagate all of them
```

Comment:
Why are these two functions split here though we seem to be calling these two functions together?

Reply:
Because they were called out at different times as pieces of logic that could be broken out. I'll combine them and change the tests.